superset-frontend/src/dashboard/components/SliceHeader/index.tsx

```diff
@@ -182,6 +183,17 @@ const SliceHeader = forwardRef<HTMLDivElement, SliceHeaderProps>(
       ({ dashboardInfo }) => dashboardInfo.crossFiltersEnabled,
     );
 
+    const chart = useSelector<RootState, Chart>(
+      state => state.charts[slice.slice_id],
```

Comment:
We need the chart object only for the first query response right? If so, can we change this selector to `state.charts[id].queriesResponse?.[0]`?
Selecting the whole chart object will trigger an unnecessary rerender every time any field in the chart changes, and we only care about the queries response here